src/service/video/docking.js

```diff
@@ -228,6 +242,97 @@ export function isElement(obj) {
 }
 
 
+/**
+ * See `src/static-template.js`.
+ * @typedef {function(!Array<string>):!Element}
+ */
+let HtmlLiteralTagDef;
+
+
+/**
+ * @param {!HtmlLiteralTagDef} html
+ * @return {!Element}
+ * @private
+ */
+const ShadowLayer = html =>
```

Comment:
Structuring the templates like this has a few benefits:

- Added readability.
- Reduced length of constructor.
- Automatic syntax highlighting in a lot of editors + plugins (by using `html` instead of `htmlFor()` directly).